Fix/moist heat capacity (#185)
contrailcirrus/pycontrails: CHANGELOG.md

### Breaking changes

- Geodesic interpolation is now used in `Flight.resample_and_fill` when the great circle distance between waypoints (rather than the total segment length including vertical displacement) exceeds a threshold. This may change the interpolation method used when resampling flight segments with lengths close to the geodesic interpolation threshold.

> - Bugfix will decrease computed values of moist heat capacity with non-zero specific humidity.

Reviewer: ```suggestion
- Fixed typo in `thermo.c_pm`  will decrease computed values of moist heat capacity with non-zero specific humidity. This update results in a negligible (<<1%) change in contrail predictions by `ISSR`, `SAC`, `PCR, and `Cocip` models.
```

Author: I'm a little hesitant to put a number on the downstream changes without actually quantifying them. The changes to values in pinned fixtures are all tiny, though, so I agree that the changes are likely to be negligible. How about
```
Fixed typo in `thermo.c_pm`  will decrease computed values of moist heat capacity with non-zero specific humidity. We expect the downstream impact on contrail predictions by `ISSR`, `SAC`, `PCR, and `Cocip` models to be negligible.
```
instead?

Reviewer: I agree - and probably best to characterize as `minimal` rather than `negligble`. Thanks!

Author: https://github.com/contrailcirrus/pycontrails/pull/185/commits/d21451bc7c60f32b9e717cb22ebc1de455254ba3

### Fixes

- Use horizontal great circle distance to determine whether geodesic interpolation is used in `Flight.resample_and_fill`. This ensures geodesic interpolation is used between sufficiently distant waypoints even when one or both waypoints contain NaN altitude data.

> - Fix moist heat capacity bug in `thermo.c_pm`.

Reviewer: ```suggestion
- Fix typo in moist heat capacity equation `thermo.c_pm`.
```

- Update Makefile so `make ensure-era5-cached` uses default cache directory when run locally.
- Bump pinned black and ruff versions.
- Disable mypy type checking on `functools.wraps` in `support_arraylike` decorator to avoid error that appears starting on mypy 1.10.0.

> - Update pinned CoCiP output values after moist heat capacity bugfix.

Reviewer: ```suggestion
- Update pinned `Cocip` test output values after moist heat capacity bugfix.
```
